packages/zero-cache/src/services/view-syncer/cvr.ts

```diff
@@ -211,8 +210,36 @@ export class CVRConfigDrivenUpdater extends CVRUpdater {
     const patches: PatchToVersion[] = [];
     const client = this.#ensureClient(clientID);
     const current = new Set(client.desiredQueryIDs);
-    const additional = new Set(queries.map(({hash}) => hash));
-    const needed = difference(additional, current);
+
+    // Find the new/changed desired queries.
+    const needed: Set<string> = new Set();
+    for (const {hash, ttl} of queries) {
+      const query = this._cvr.queries[hash];
+      if (!query) {
+        needed.add(hash);
+        continue;
+      }
+      if (query.internal) {
+        continue;
+      }
+
+      const oldClientState = query.clientState[clientID];
+      // Old query was inactivated or never desired by this client.
+      if (!oldClientState || oldClientState.inactivatedAt !== undefined) {
+        needed.add(hash);
+        continue;
+      }
+
+      if (
+        ttl !== oldClientState.ttl &&
+        oldClientState.ttl !== undefined &&
+        ttl !== undefined &&
+        ttl >= oldClientState.ttl
+      ) {
+        needed.add(hash);
+      }
```

Comment:
What if the old client state had no ttl but the new state does have a ttl? Does the query record get revised with the new TTL even though it is not added to the `needed` set?